### PR TITLE
feat(docs): new sdk documentation template

### PR DESCRIPTION
**Pull request by pedronauck**
## Tasks

- [x] Setup NextJS with MDX to auto-generate pages
- [x] Integrate DocSearch
- [x] Auto scrap Docsearch using Github Action
- [x] Deploy on Vercel
- [x] Fix `@fuel-ui` NextJS integration

### CODE REVIEW

**Review by luizstacio — CHANGES_REQUESTED** [review]
- [x] Increase contrast on texts to improve readability
- [x] Add the video on install instructions
- [x] Fix layout responsiveness  
- [ ] Replace code snippets on the md with real code snippets
   - [ ] Move the code examples to separated files
   - [ ] Move the tests of the code snippets to the side of this files
   - [ ] Add the code snippet import on the markdown files.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
       - name: Bump and Collect Version
         run: |
           pnpm changeset version
-          echo "BUILD_VERSION=$(sed -nE 's/^\s*"version": "(.*?)",$/\1/p' design-system/react/package.json)" >> $GITHUB_ENV
+          echo "BUILD_VERSION=$(sed -nE 's/^\s*"version": "(.*?)",$/\1/p' packages/app/package.json)" >> $GITHUB_ENV
```
Comment (luizstacio) [inline .github/workflows/release.yaml:36]: We should split this fixes in small PRs: https://github.com/FuelLabs/fuels-wallet/pull/255

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Generated by Cargo
 # will have compiled files and executables
 /target/
+/out
+packages/docs/.vscode
```
Comment (luizstacio) [inline .gitignore:5]: Do we need this?

File: packages/docs/.env.example [hunk 3]
```diff
@@ -0,0 +1,6 @@
+NEXT_PUBLIC_ALGOLIA_APP_ID=
+NEXT_PUBLIC_ALGOLIA_SEARCH_API_KEY=
+ALGOLIA_SEARCH_ADMIN_KEY=
```
Comment (luizstacio) [inline packages/docs/.env.example:3]: We can remove `ALGOLIA_SEARCH_ADMIN_KEY` from the example as it is used only on the CI

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: scripts/gh-pages-preview.mjs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { join } from "path";
```
Comment (luizstacio) [inline scripts/gh-pages-preview.mjs:1]: Just as a note, this refactoring could be done on a different PR from the docs PR.

File: packages/docs/src/lib/str.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export function capitalize(str: string) {
+  return `${str.slice(0, 1).toUpperCase()}${str.slice(1, Infinity)}`;
```
Comment (luizstacio) [inline packages/docs/src/lib/str.ts:2]: Why pass Infinity here?
```suggestion
  return `${str.slice(0, 1).toUpperCase()}${str.slice(1)}`;
```

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (pedronauck) [reply .github/workflows/release.yaml:36]: I just fix because I saw it

File: .gitignore (hunk 2 above)
Comment (pedronauck) [reply .gitignore:5]: Next13 it's auto-generating this automatically, I could see how to disable this option. So, I just ignored it.

File: packages/docs/.env.example (hunk 3 above)
Comment (pedronauck) [reply packages/docs/.env.example:3]: ✅

File: scripts/gh-pages-preview.mjs (hunk 4 above)
Comment (pedronauck) [reply scripts/gh-pages-preview.mjs:1]: Noted

File: packages/docs/src/lib/str.ts (hunk 5 above)
Comment (pedronauck) [reply packages/docs/src/lib/str.ts:2]: I didn't know that by omitting the second parameter it got it the rest, cool!

**Review by luizstacio — APPROVED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**